colcon_parallel_executor/executor/parallel.py

```diff
@@ -131,8 +140,9 @@ async def _execute(self, args, jobs, *, on_error):
             for package_name, job, _ in ready_jobs:
                 # don't schedule more jobs then workers
                 # to prevent starting further jobs when a job fails
-                if len(futures) + len(take_jobs) >= args.parallel_workers:
-                    break
+                if args.parallel_workers:
+                    if len(futures) + len(take_jobs) >= args.parallel_workers:
+                        break
```

Comment:
It looks intentional that this conditional _won't_ run when `args.parallel_workers` is 0 (unlimited) but I did have to stop and think it through. I don't think there's anything actionable I'm just reviewing "out loud".

Reply:
A `break` in this loop causes the executor to stop looking for new work. Skipping the conditional (and the possible break) means that the executor will pluck all available work and begin executing on it regardless of the `parallel_workers` value.

I find the logic in this loop _very_ hard to follow, so I understand the pause this change yields upon review.